src/tiledb/cloud/_common/utils.py

```diff
@@ -21,18 +20,85 @@
 logger = logging.getLogger("tiledb.cloud")
 
 
-def split_uri(uri):
+def canonicalize_ns_name_uri(
+    **ns_name_uri: Optional[str],
```

Comment:
Although this is clever, should we do this more strict and just have the `namespace`, `name` and `uri` there?

Reply:
Whoops! Forgot to respond to this:

I’m not hugely worried about this because it’s internal-only, and while this is a little clever, it makes the callsites extremely straightforward, since the call is always going to look like `param_name=param_name`.